### ISSUE TITLE
Sometimes the changelog generated is very strange

### ISSUE THREAD

**Issue by syrm** [bvieira/sv4git#16]
Hello,

Try to generate the changelog of this repository `https://github.com/syrm/ValheimTwitch`
I'm using `git sv cgl`

The result is strange, as you can see : 
```
# Changelog

## v1.3.0 (2021-03-27)

### Features

- Viewer can change weather (73efa23)
- Start hord event (74d07ef)
- Add should redemptions skip request queue option (e018e4a)

---

## v1.2.0 (2021-03-26)

### Features

- Enable/Disable reward when game quit (2eeb738)
- Enable/Disable reward when game state change (b8184ea)
- Follow my channel to get rid of intro message (f3d7fe3)
- Add reward form (454be83)

---

## v1.1.0 (2021-03-25)

### Features

- Persistant rewards settings (595d749)
- Rewards configuration (1669852)

---

## v1.0.0 (2021-03-24)

---

## v1.4.2

---

## v1.5.0

---

## v1.5.1

### Features

- Add shortcuts (#17) (63f7cf4)
- Update version notification (#12) (43824a3)
- Show Hugin intro only once per game session (#11) (b20cab6)
- Spawn boss (not tameable) (#10) (9808107)
- Spawn Tamed creature which takes the name of the viewer (#9) (a25bc21)
- Player heal and puke (#8) (58ad18c)
- Switch to edit action right after creation (7a782ac)
- Disable reward at creation (ac70130)
- Viewer can change weather (73efa23)
- Start hord event (74d07ef)
- Add should redemptions skip request queue option (e018e4a)
- Enable/Disable reward when game quit (2eeb738)
- Enable/Disable reward when game state change (b8184ea)
- Follow my channel to get rid of intro message (f3d7fe3)
- Add reward form (454be83)
- Persistant rewards settings (595d749)
- Rewards configuration (1669852)
- Set reward action (c0b9eb0)
- Reward settings panel base (9ad4774)
- Refresh rewards list (8778941)
- New GUI (db2a277)
- New GUI (POC) (e2c085a)
- Embeded GUI (POC) (7cb433d)
- Spawn board hord (06e7543)
- Spawn Troll (7f9bbe4)
- Print HUD messages (top-left/center) (a72caf3)
- Spawn Hugin/Munin (0fd9827)
- Run abstract action (a736363)
- Save reward/action in user prefs (9b3094d)
- CustomDropdown (988a062)
- CustomDropdown (742950c)
- Rewards grid (54b0f2b)
- VerticalScrollView (6e83ac7)
- Twitch refresh token (8d2cb1a)
- Twitch login (71bc753)
- MainGui + MainButton (poc) (a15e517)
- Reawrds grid (poc) (6b569fa)
- ScollView (311b9ee)
- Settings panel container (e120b73)
- Embedded assemblies (1d62abf)
- Get rewards list (7b76af3)
- Better login button (00608f9)
- Twitch login (226646d)
- Get twitch user token (7e92bec)
- OnAuthCode event (6fb0ff6)
- HTML response (ae479db)
- Get Twitch auth code (da4f57d)
- Login button image (c906618)
- Twitch login button (draft) (9d682a8)
- PubSub auto reconnect (812bd70)
- JSON.NET Serialization (a7d2102)
- PubSub POC (8a44240)
- Startup message (d357d8b)
- Get Twitch user as TwitchUser type (6fdc5fc)
- Get Twitch user (7844113)
- Base config (d1248e3)

### Bug Fixes

- Tamed creature faction (#23) (ff98724)
- Reward form setup (#22) (f11d403)
- Only tamed creature is put into player faction (#21) (b52fa84)
- Only tamed creature is put into player faction (#18) (33eef56)
- Undefined tamed property (#14) (ac26ec8)
- Check channel points availability (431d27c)
- Clear add new reward form before open (9ec8ac6)
- Deletes a reward when it no longer exists (ec795b1)
- Reward form toggle not taken into account (825697e)

---

```

Some tags are empty, and the order of tag is wrong.

**Comment by bvieira** [comment]
Hi,

`sv4git` works with annotated tags, it seems that you stopped to use annotated tags and started to use lightweight instead. Is there any reason for that?

```bash
$ git for-each-ref refs/tags  
a47ef71c68fbdcc3a85bb1d42518c3e81cc7f1e7 tag	refs/tags/v1.0.0-beta
bff00c1342bd1122c6687a17090abc45ea8373c6 tag	refs/tags/v1.1.0-beta
84110f85855352dcbf9d10fe4d61689bbf5165c7 tag	refs/tags/v1.2.0
d25cf252fc7a0ec72719d798575fe1b46dceedcf tag	refs/tags/v1.3.0
731595726eeb7f184cf36c40cd822f17d89b6e5b commit	refs/tags/v1.4.2
0ec75357e695e7e19691c1397eb7d02034ef7c23 commit	refs/tags/v1.5.0
61c1e9adc6e37e92a974f32a6f156067427509b7 commit	refs/tags/v1.5.1
```

The `taggerdate` is used to sort tags from repo, and lightweight tags don't have this date

```bash
$ git tag -l --format "%(taggerdate:iso8601) %(refname:short)"
2021-03-24 09:28:09 +0100 v1.0.0-beta
2021-03-25 14:16:36 +0100 v1.1.0-beta
2021-03-26 18:05:52 +0100 v1.2.0
2021-03-27 18:02:57 +0100 v1.3.0
 v1.4.2
 v1.5.0
 v1.5.1
```

It's not only `git sv cgl` that it's not working, `current-version`, `next-version` and `tag` are broken as well, because `git describe` get the last annotated tag (v1.3.0).

```bash
$ git sv cv
1.3.0
$ git sv nv
1.4.0
```

I think that I can change it to support both tags (annotated and lightweight), but when using lightweight tags, the date from your tag is the commit date, so your release date will be the same as the last commit and not the date the tag was created. Eg.:

| last commit date | tag date | tag type | tag | release date |
| -- | -- | -- | -- | -- |
| 2021-02-25 21:37:33 -0300 | 2021-04-10 15:58:45 -0300 | lightweight | v1.0.0 | 2021-02-25 |
| 2021-02-25 21:37:33 -0300 | 2021-04-10 15:58:45 -0300 | annotated | v1.0.0 | 2021-04-10 |

**Comment by syrm** [comment]
Thank for theses informations.
I think support both and take date of commit for lightweight tag seems good.

**Comment by bvieira** [comment]
I forgot about the empty releases, currently `sv4git` only shows feat, fix and breaking changes on changelog, if your tag don't have any, it will be blank :(

In the future I would like to add a misc section to changelog, but I don't know when..

```bash
$ git sv cl -t v1.5.1                                          
{"date":"2021-04-10","hash":"61c1e9a","message":{"type":"chore","description":"v1.5.1 (#27)"}}
{"date":"2021-04-10","hash":"3216a77","message":{"type":"chore","description":"Add build tools (#26)"}}
```

**Comment by syrm** [comment]
Indeed i noticed this :-)

Maybe in configuration allows the user to add the headers he wants for each type, if he doesn't add the type in headers, dont show it in changelog.

**Comment by bvieira** [comment]
> Indeed i noticed this :-)
> 
> Maybe in configuration allows the user to add the headers he wants for each type, if he doesn't add the type in headers, dont show it in changelog.

yeah, and some way to aggregate more than one type as well.. but I need to make headers flexible on template first..

**Comment by bvieira** [comment]
support for lightweight tags is available in version 2.2.0

**Comment by bvieira** [comment]
support for lightweight tags is available in version 2.2.0

**Comment by syrm** [comment]
Thanks ! As always you're awesome :-)
